### PR TITLE
Update schemas so that they use Java 8 optionals instead of Guava

### PR DESCRIPTION
**Pull request by amarzoca**
Goals (and why):
Fix schemas that require Java 8 Optionals.

Implementation Description (bullets):
* Change the schema code so that it uses OptionalType.JAVA8. 
* Regenerate relevant classes.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2210)

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/atlasdb](https://github.com/palantir/atlasdb), @amarzoca! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by tpetracca** [comment]
This might just be exposing an existing bug, but it feels weird to me that after specificing `OptionalType.JAVA8` a bunch of our rendered code went from using java optionals to using guava optionals...
Example from RangeScanTestTable:
```
  import java.util.List;
  import java.util.Map;
  import java.util.Map.Entry;
 -import java.util.Optional;
  import java.util.Set;
  import java.util.SortedMap;
  import java.util.concurrent.Callable;
 @@ -20,6 +19,7 @@
  import com.google.common.base.Joiner;
  import com.google.common.base.MoreObjects;
  import com.google.common.base.Objects;
 +import com.google.common.base.Optional;
  import com.google.common.base.Supplier;
  import com.google.common.collect.ArrayListMultimap;
  import com.google.common.collect.Collections2;
 @@ -479,7 +479,7 @@ public void delete(Iterable<RangeScanTestRow> rows) {
          byte[] bytes = row.persistToBytes();
          RowResult<byte[]> rowResult = t.getRows(tableRef, ImmutableSet.of(bytes), columns).get(bytes);
          if (rowResult == null) {
 -            return Optional.empty();
 +            return Optional.absent();
          } else {
              return Optional.of(RangeScanTestRowResult.of(rowResult));
          }

```

**Comment by amarzoca** [comment]
That's actually expected as the default in the [Schema class](https://github.com/palantir/atlasdb/blob/develop/atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/Schema.java) (line 95) was Guava and I left it unchanged. I regenerated all the schemas but only added Java8 to those that broke without it, so some regenerated with Guava.
Do you think we should add Java8 Optionals as the default?

File: examples/profile-client/src/main/java/com/palantir/example/profile/schema/generated/UserProfileTable.java [hunk 1]
```diff
@@ -1337,9 +1337,9 @@ public int hashCode() {
             }
 
             @Override
-            public int compareTo(CookiesIdxRow cookiesIdxRow) {
```
Comment (hsaraogi) [inline examples/profile-client/src/main/java/com/palantir/example/profile/schema/generated/UserProfileTable.java]: seems odd that this changed? Was there a change in the Schema definition?

File: examples/profile-client/src/main/java/com/palantir/example/profile/schema/generated/UserPhotosStreamStore.java [hunk 2]
```diff
@@ -219,7 +219,7 @@ protected void loadSingleBlockToOutputStream(Transaction t, Long streamId, long
                 Long streamId = v.getColumnName().getStreamId();
                 Sha256Hash hash = r.getHash();
                 if (hashForStreams.containsKey(streamId)) {
-                    AssertUtils.assertAndLog(log, hashForStreams.get(streamId).equals(hash), "(BUG) Stream ID has 2 different hashes: " + streamId);
+                    AssertUtils.assertAndLog(hashForStreams.get(streamId).equals(hash), "(BUG) Stream ID has 2 different hashes: " + streamId);
```
Comment (hsaraogi) [inline examples/profile-client/src/main/java/com/palantir/example/profile/schema/generated/UserPhotosStreamStore.java]: I guess we added this logger recently, we might have added it to the gen code but not to schema renderer, can you add it to the schema renderer?

Comment (amarzoca) [inline examples/profile-client/src/main/java/com/palantir/example/profile/schema/generated/UserPhotosStreamStore.java]: Fixed the StreamStore renderer to use that logger.

File: examples/profile-client/src/main/java/com/palantir/example/profile/schema/generated/UserProfileTable.java (hunk 1 above)
Comment (amarzoca) [inline examples/profile-client/src/main/java/com/palantir/example/profile/schema/generated/UserProfileTable.java]: No, there wasn't, I just regenerated the classes. The change is due to a previous PR where some linting was applied that changed the param name in the generated code.

**Pull request merged**